cmake/modules/FindCMock.cmake

```diff
@@ -0,0 +1,45 @@
+# SPDX-FileCopyrightText: Silicon Laboratories Inc. <https://www.silabs.com/>
+# SPDX-License-Identifier: Zlib
+#
+# This recipe allows to download CMock
+# It can be used by projects which are depending on it
+# Feel free to copy this (up to date) file everywhere it is needed
+
+include(FetchContent)
+
+if(NOT DEFINED CMOCK_GIT_REPOSITORY)
+  if(DEFINED ENV{CMOCK_GIT_REPOSITORY})
+    set(CMOCK_GIT_REPOSITORY $ENV{CMOCK_GIT_REPOSITORY})
+  endif()
+endif()
+if("${CMOCK_GIT_REPOSITORY}" STREQUAL "")
+  set(CMOCK_GIT_REPOSITORY "https://github.com/ThrowTheSwitch/CMock")
+endif()
+
+if(NOT DEFINED CMOCK_GIT_TAG)
+  if(DEFINED ENV{CMOCK_GIT_TAG})
+    set(CMOCK_GIT_TAG $ENV{CMOCK_GIT_TAG})
+  else()
+    set(CMOCK_GIT_TAG "v2.5.3")
```

Comment:
Maybe we should also align the version in our dependency which is UnifySDK:

https://github.com/SiliconLabs/UnifySDK/tree/fb360ad38cdc677ba1859abd04fe3e4e12d53a65/components/testframework/libs/cmock